### PR TITLE
Figure out how to "upload" resources to GitHub

### LINKED ISSUE #7: Figure out how to "upload" resources to GitHub
When a user picks their photos and specifies their project's metadata #6 , we need to "upload" it to the new GitHub repo that we build for them (#1).  We'll have to make a commit to git on their behalf, and `push` to GitHub, or somehow use the GitHub API to create a commit with all the necessary files.

Once this commit is made, the GitHub Actions (#3 and #5) should kick in and create everything.

### PR DESCRIPTION
**Pull request by saminarp**
Fixes #7 

I have done my implementation based on PRs created for https://github.com/humphd/my-photohub/issues/1 and https://github.com/humphd/my-photohub/issues/4 

Please review and let me know if this is the right approach. Feel free to review and request for changes.

### CODE REVIEW

File: github-upload.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Octokit, App } from "https://cdn.skypack.dev/octokit?dts";
```
Comment (batunpc) [inline github-upload.js:1]: App does not need to be imported

File: github-upload.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return await octokit.request('PUT /repos/{owner}/{repo}/contents/{path}', {
+        owner,
+        repo,
+        path: `${UPLOAD_DIRECTORY}/${file.name}`,
+        message: `Upload File - ${file.name}`,
+        content: await getBase64(file)
+    })
+}
+
+window.githubUpload = githubUpload
```
Comment (batunpc) [inline github-upload.js:26]: This line seems redundant.

**Review by batunpc — CHANGES_REQUESTED** [review]
@saminarp I have left some review regarding the unused code. 

Other than that I think it looks good! 😀

File: github-upload.js (hunk 1 above)
Comment (LostButton) [inline github-upload.js:1]: Would it not be better practice to import Octokit via node_modules?  I get that this works, but apparently importing from HTTP is still experimental.
https://nodejs.org/api/esm.html#https-and-http-imports

**Review by LostButton — CHANGES_REQUESTED** [review]
Honestly, this is all I have to say, otherwise, it looks good!

**Comment by SerpentBytes** [comment]
@saminarp  Given that you coded the functions elegantly, I did not want to point out that `App` was imported but not used because it was too minor. Good work.

**Comment by saminarp** [comment]
I have updated my implementation, let me know how it is

File: github-upload.js (hunk 1 above)
Comment (saminarp) [reply github-upload.js]: Since my file was pretty small, I did it this way. I am not sure if there are any restrictions.

**Review by batunpc — APPROVED** [review]
Thanks @saminarp, it looks good

**Review by LostButton — APPROVED** [review]
This works

**Review by SerpentBytes — APPROVED** [review]
LGTM

**Pull request merged**